python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -376,8 +389,20 @@ class NearestNeighbors(Base):
         cdef uintptr_t I_ptr = get_dev_array_ptr(I_ndarr)
         cdef uintptr_t D_ptr = get_dev_array_ptr(D_ndarr)
 
-        cdef float** inputs = <float**><size_t>self.input
-        cdef int* sizes = <int*><size_t>self.sizes
+        cdef float** inputs
+        cdef int* sizes
+        cdef uintptr_t In_ctype
+        if self.n_indices > 1:
+            inputs = <float**><size_t>self.input
+            sizes = <int*><size_t>self.sizes
+        else:
+            In_ctype = get_dev_array_ptr(self.X_m)
+
+            inputs = <float**> malloc(sizeof(float *))
```

Comment:
Where do these get freed?

Reply:
Great catch! Looks like I removed the `__dealloc__` in the stateless Python KNN PR and didn't add it to the new Python class. 

https://github.com/rapidsai/cuml/pull/683/files#diff-cbd7000fb11a0e05114d7da53412c9f8L120

Reply:
Fixed this in 0.8 hotfix. Will merge into this branch soon.